ament_uncrustify/ament_uncrustify/main.py

```diff
@@ -106,106 +111,22 @@ def main(argv=sys.argv[1:]):
-            output_files = [
-                os.path.join(temp_path, f + suffix) for f in input_files]
-        else:
-            # input files are absolute
-            # remove leading slash, prepend temp path, append suffix
-            output_files = [
-                os.path.join(
-                    temp_path,
-                    os.sep.join(f.split(os.sep)[1:]) +
-                    suffix
-                ) for f in input_files
-            ]
-
-        uncrustified_files = output_files
-        i = 1
-        while True:
-            # identify files which have changed since the latest uncrustify run
-            changed_files = []
-            for input_filename, output_filename in zip(
-                    input_files, uncrustified_files):
-                if cwd and not os.path.isabs(input_filename):
-                    input_filename = os.path.join(cwd, input_filename)
-                if not filecmp.cmp(input_filename, output_filename):
-                    if output_filename == input_filename + suffix:
-                        # for repeated invocations
-                        # replace the previous uncrustified file
-                        os.rename(output_filename, input_filename)
-                        changed_files.append(input_filename)
-                    else:
-                        # after first invocation remove suffix
-                        # otherwise uncrustify behaves different
-                        output_filename_without_suffix = \
-                            output_filename[:-len(suffix)]
-                        os.rename(
-                            output_filename, output_filename_without_suffix)
-                        changed_files.append(output_filename_without_suffix)
-            if not changed_files:
-                break
-            # reinvoke uncrustify for previously changed files
-            input_files = changed_files
-            try:
-                cmd = [uncrustify_bin,
-                       '-c', args.config_file,
-                       '--suffix', suffix]
-                cmd.extend(input_files)
-                subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
-            except subprocess.CalledProcessError as e:
-                if e.output:
-                    print(e.output, file=sys.stderr)
-                print("The invocation of 'uncrustify' failed with error code %d: %s" %
-                      (e.returncode, e), file=sys.stderr)
-                return 1
-
-            uncrustified_files = [f + suffix for f in input_files]
-            i += 1
-            if i >= 5:
-                print("'uncrustify' did not settle on a final result even "
-                      "after %d invocations" % i, file=sys.stderr)
-                return 1
+        all_input_files = []
+        all_output_files = []
+        for language, input_files in files_by_language.items():
+            input_files = files_by_language[language]
```

Comment:
@dirk-thomas unnecessary?

Reply:
Can you please elaborate what you mean.

Reply:
There's no need to retrieve `input_files` again from the dictionary that is being iterated.

Reply:
Ah, now I got it: removed in c330e7d8c834a83f87ff69d59d93661433cd0c38.